archinfo/arch.py

```diff
@@ -1,8 +1,12 @@
 import logging
+from typing import Dict, List, Tuple
+
 import struct as _struct
 import platform as _platform
 import re
 from archinfo.archerror import ArchError
+from pyvex import RegisterOffset, RegisterName
```

Comment:
We shouldn't be using pyvex in archinfo.

Reply:
This raises the issue of where to put those type aliases. They need to be unique, so it is not possible for each module to define it themselves. The only other option that comes to mind right now would be a dedicated module for type aliases like this. Declaring that some integer is a RegisterOffset and not a TempVar is one of the most useful benefits of type annotations for this part of the codebase, so I definitely want this somewhere.
The meaning mapping registername <-> register offset comes from Vex it seems reasonable to me to get it from pyvex. What issues would be introduced by this?

Reply:
archinfo just needs to still work when pyvex isn't installed, just without the pyvex features. you can put this in an optional idiom and make them None if pyvex throws an ImportError.

Alternately, the type aliases can just live in archinfo.